### PR TITLE
docs(patch): Add "About Percentiles" page

### PR DESCRIPTION
**Pull request by finestructure**
## Description

Adds a new documentation page about percentiles and how to interpret them. 

## How Has This Been Tested?

Generated the docs locally in Xcode.

## Minimal checklist:

- [x] I have performed a self-review of my own code 
- [x] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by finestructure** [comment]
@hassila , as discussed I've taken a stab at extending the docs with a little more info about percentiles and the like.

I feel like the last section "Interpreting results" could have a little more to it but I struggled to come up with anything other than what I've got. I recall from watching Gil Tene's talks that there should be something about focusing on the worst value but I'm unsure how to phrase that (or even if that's always the best thing to do).

Hope that helps regardless!

**Comment by hassila** [comment]
This is awesome, thanks @finestructure - much appreciated!

File: Sources/Benchmark/Documentation.docc/Resources/Images/PercentileHistogramExample.png
Comment (finestructure) [inline Sources/Benchmark/Documentation.docc/Resources/Images/PercentileHistogramExample.png:1]: I pulled this file from http://hdrhistogram.github.io/HdrHistogram and I believe it has a Creative Commons license. I believe all it needs is attribution, maybe just "from: https://hdrhistogram.github.io/HdrHistogram/". Or maybe it's better to recreate it. I can try and do that if you prefer.

Comment (hassila) [reply Sources/Benchmark/Documentation.docc/Resources/Images/PercentileHistogramExample.png:1]: Could perhaps recreate an image using the plotter (http://hdrhistogram.github.io/HdrHistogram/plotFiles.html) - then we don't just lift the resource as-is?

Comment (finestructure) [reply Sources/Benchmark/Documentation.docc/Resources/Images/PercentileHistogramExample.png:1]: Yes, I was thinking that, too, the problem is getting a sensible data set. I'll try to generate something but if you have something to hand, that'd be great!

Comment (hassila) [reply Sources/Benchmark/Documentation.docc/Resources/Images/PercentileHistogramExample.png:1]: I'd just go with sample data set they have there if you open the link, I think that's fine.

Comment (finestructure) [reply Sources/Benchmark/Documentation.docc/Resources/Images/PercentileHistogramExample.png:1]: Ah nice, the example data is right there. Updated!

**Comment by hassila** [comment]
Very nice, thanks again!

**Review by hassila — APPROVED** [review]

**Pull request merged**